### PR TITLE
JSON-RPC Cache improvements (authorization in dispatcher, poll error metric fix)

### PR DESCRIPTION
**Pull request by dkeysil**
- I'm adding support of JWT authorization on dispatcher in blocksdata client
- Poll error metric should be created as system metric

### CODE REVIEW

File: clients/blocksdata/r2_client.go [hunk 1]
```diff
@@ -54,22 +58,34 @@ func (c *blocksDataClient) GetBlocksData(bucket int64) (_ *protocol.BlocksData,
...
 		}
 
-		defer resp.Body.Close()
+		jwt, err := security.CreateScannerJWT(c.key, nil)
+		if err != nil {
+			return backoff.Permanent(err)
+		}
 
-		b, err := io.ReadAll(resp.Body)
+		req.Header.Set("Authorization", "Bearer "+jwt)
```
Comment (aomerk) [inline clients/blocksdata/r2_client.go:71]: can we use `fmt.Sprintf(..)` here?

**Review by aomerk — APPROVED** [review]

File: clients/blocksdata/r2_client.go [hunk 2]
```diff
@@ -54,22 +58,34 @@ func (c *blocksDataClient) GetBlocksData(bucket int64) (_ *protocol.BlocksData,
...
-		defer resp.Body.Close()
+		scannerJwt, err := security.CreateScannerJWT(c.key, nil)
+		if err != nil {
+			return backoff.Permanent(err)
+		}
 
-		b, err := io.ReadAll(resp.Body)
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", scannerJwt))
+
+		resp, err := httpclient.Default.Get(dispatcherUrl)
```
Comment (aomerk) [inline clients/blocksdata/r2_client.go:73]: no one uses the request....

Comment (aomerk) [reply clients/blocksdata/r2_client.go:73]: it should be `http.DefaultClient.Do(req)`

Comment (canercidam) [reply clients/blocksdata/r2_client.go:73]: (Please note the difference between `httpclient.Default` and `http.DefaultClient` - the former is from our core repo and has some defaults.)

Comment (aomerk) [reply clients/blocksdata/r2_client.go:73]: yes, the constructed request is not used and therefore the requests don't have the auth bearer header - client is making a generic GET call to dispatcher

**Pull request merged**